src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs

```diff
@@ -547,9 +563,19 @@ private void ReadEndOfCentralDirectory()
 
                 _expectedNumberOfEntries = eocd.NumberOfEntriesInTheCentralDirectory;
 
-                // only bother saving the comment if we are in update mode
-                if (_mode == ZipArchiveMode.Update)
+                if (eocd.ArchiveComment.Length > 0)
+                {
                     _archiveComment = eocd.ArchiveComment;
+                    _archiveCommentAsString = ZipHelper.TruncateEncodedBytesAndReturnString(
```

Comment:
What is the advantage of converting bytes to string now? 

If this logic was part of `Comment` getter, only the users who actually use it would perform the extra work.

Reply:
Changed.